Subtle Alert Style (#773)
smartdevicelink/sdl_evolution: proposals/nnnn-alert-style-subtle.md

# Subtle Alert Style

* Proposal: [SDL-NNNN](NNNN-alert-style-subtle.md)
* Author: [Joel Fischer](https://github.com/joeljfischer)
* Status: **Awaiting review**
* Impacted Platforms: [Core / iOS / Java Suite / HMI / RPC]

## Introduction
This feature adds a new style of alert: the "subtle" alert, which does not take over the entire screen. The current alerts are moved to a style called "prominent."

## Motivation
Competitive systems in this space often have "notification" style alerts, and user expectations are often set by mobile devices, which have notification alerts that don't take over the screen. To meet user and developer expectations, we should provide both prominent (modal) and subtle (notification) style alerts.

> Places where subtle alerts may be useful is for messaging apps that notify the user of a new message, a VOIP app notifying the user of an incoming call, or a navigation app notifying the user of an upcoming accident. These should have the ability to be presenting outside of the current screen-takeover method.

Reviewer: ```suggestion
Places where subtle alerts may be useful include messaging apps that notify the user of a new message, a VOIP app notifying the user of an incoming call, or a navigation app notifying the user of an upcoming accident. These should have the ability to be presented outside of the current screen-takeover method.
```

## Potential downsides
Because this is an addition to the `Alert` RPC and not a new separate RPC, policies cannot allow an app to send a "subtle" notification while preventing them from sending a "prominent" notification. This is a significant downside and almost caused the author to change the approach. The author ultimately determined that adding a new non-mandatory parameter was better due to the API similarity, but if the SDLC decides to use Alternative 1 below, that would be acceptable by the author.

## Impact on existing code
This would be a minor version change.

## Alternatives considered

### Alternative 1: Separate RPC
Using a separate RPC allows us to (1) control this alert style separately from `Alert` in policies, and (2) to provide a callback for when the notification is touched. This would permit better control for the app developers than simply opening the app.

```xml
<function name="SubtleAlert" functionID="SubtleAlertID" messagetype="request">
    <description>Shows an alert which typically consists of text-to-speech message and text on the display. At least either alertText1, alertText2 or TTSChunks need to be provided.</description>
    

>     <description>Shows an alert which typically consists of text-to-speech message and text on the display. At least either alertText1, alertText2 or TTSChunks need to be provided.</description>

Reviewer: Is this description supposed to be here twice?

Author: 🤔 I don't know how that happened
